test/Utils.hs

```diff
@@ -66,13 +66,14 @@ printProgramResult = printProcResult . fromProgramResult
 type AgdaArgs = [String]
 
 
-readAgdaProcessWithExitCode :: AgdaArgs -> Text
+readAgdaProcessWithExitCode :: Maybe [(String, String)]
+                            -> AgdaArgs -> Text
                             -> IO (ExitCode, Text, Text)
-readAgdaProcessWithExitCode args inp = do
+readAgdaProcessWithExitCode env args inp = do
```

Comment:
Should the following `getEnvAgdaArgs` take `env` into account?
Either way, please leave a comment _in the code_ clarifying this.

Reply:
Made `getEnvAgdaArgs` & related helpers pure